### ISSUE TITLE
PHP warning when running the latest phar

### ISSUE THREAD

**Issue by ivokund** [squizlabs/PHP_CodeSniffer#944]
This started happening probably after our CI service started downloading the new 2.6.0 release. 

Running this:
```
url -OL https://squizlabs.github.io/PHP_CodeSniffer/phpcs.phar
php phpcs.phar --standard=PSR2 src 
```
Produces:
```
PHP Warning:  chdir(): No such file or directory (errno 2) in phar:///media/sf__SHARE/htdocs/adsales/phpcs.phar/CodeSniffer.php on line 861
```

Note that the described error does not appear if running without the `--standard=PSR2` flag. Also, running a composer version of PHPCS does not give the error.

**Comment by aik099** [comment]
What version of PHP do you use?

**Comment by ivokund** [comment]
Sorry, forgot to mention: PHP 7.0.1-5+deb.sury.org~trusty+1

**Comment by voult** [comment]
Can confirm on:
- PHP 7.0.4 (win)
- CodeSniffer 2.6.0
- NetBeans 8.0.2 (using "phpCS-MD" plugin)

While setting standard to "PSR2".

Here is dump from manual run with the same settings as NetBeans does:

>php phpcs.phar [some_path_to_tested_script_here] --standard=PSR2 --report=xml --encoding=UTF8

PHP Warning:  chdir(): No such file or directory (errno 2) in phar://[some_path_here]/phpcs.phar/CodeSniffer.php on line 861

Warning: chdir(): No such file or directory (errno 2) in phar://[some_path_here]/phpcs.phar/CodeSniffer.php on line 861

PHP Notice:  iconv(): Wrong charset, conversion from `utf8' to `utf-8' is not allowed in phar://[some_path_here]/phpcs.phar/CodeSniffer/Reports/Xml.php on line 76

PHP Notice:  iconv(): Wrong charset, conversion from `utf8' to `utf-8' is not allowed in phar://[some_path_here]/phpcs.phar/CodeSniffer/Reports/Xml.php on line 76

<?xml version="1.0" encoding="UTF-8"?>
<phpcs version="2.6.0">

Notice: iconv(): Wrong charset, conversion from `utf8' to `utf-8' is not allowed in phar://[some_path_here]/phpcs.phar/CodeSniffer/Reports/Xml.php on line 76

Notice: iconv(): Wrong charset, conversion from `utf8' to `utf-8' is not allowed in phar://[some_path_here]/phpcs.phar/CodeSniffer/Reports/Xml.php on line 76

<file name="[some_path_to_tested_script_here]" errors="2" warnings="0" fixable="2">
 <error line="68" column="32" source="Squiz.WhiteSpace.ControlStructureSpacing.SpacingAfterOpen" severity="5" fixable="1"></error>
 <error line="71" column="1" source="Squiz.WhiteSpace.ControlStructureSpacing.SpacingBeforeClose" severity="5" fixable="1"></error>
</file>
</phpcs>

**Comment by mkasberg** [comment]
I am experiencing the same PHP warning on:

* PHP 5.6.11
* CodeSniffer 2.6.0

It only happens if the `--standard=foo` argument is passed.

**Comment by mkasberg** [comment]
I pulled down master, and found some interesting behavior. I cannot reproduce the issue when I use the phpcs script in my repository. But I can reproduce it with the phar I have installed. Also, here's a stacktrace:

    $ pwd
    .../PHP_CodeSniffer/scripts
    
    $ which phpcs
    /usr/local/bin/phpcs
    
    $ phpcs --standard=Custom-PHP ../../code/www/index.html 
    PHP Warning:  chdir(): No such file or directory (errno 2) in phar:///usr/local/bin/phpcs/CodeSniffer.php on line 861
    PHP Stack trace:
    PHP   1. {main}() /usr/local/bin/phpcs:0
    PHP   2. PHP_CodeSniffer_CLI->runphpcs() /usr/local/bin/phpcs:6
    PHP   3. PHP_CodeSniffer_CLI->process() phar:///usr/local/bin/phpcs/CodeSniffer/CLI.php:104
    PHP   4. PHP_CodeSniffer->initStandard() phar:///usr/local/bin/phpcs/CodeSniffer/CLI.php:870
    PHP   5. PHP_CodeSniffer->processRuleset() phar:///usr/local/bin/phpcs/CodeSniffer.php:561
    PHP   6. PHP_CodeSniffer->_expandRulesetReference() phar:///usr/local/bin/phpcs/CodeSniffer.php:757
    PHP   7. PHP_CodeSniffer->processRuleset() phar:///usr/local/bin/phpcs/CodeSniffer.php:1150
    PHP   8. chdir() phar:///usr/local/bin/phpcs/CodeSniffer.php:861

    $ ./phpcs --standard=Custom-PHP ../../code/www/index.html 
    <NO PHP WARNING>

The issue does not appear to depend on which file is being analyzed.

**Comment by mkasberg** [comment]
This was probably introduced by #847.

**Comment by sebastianbergmann** [comment]
Have the same issue:
```
$ php --version
PHP 7.0.5 (cli) (built: Apr  8 2016 17:03:20) ( NTS )
Copyright (c) 1997-2016 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2016 Zend Technologies
    with Xdebug v2.4.0, Copyright (c) 2002-2016, by Derick Rethans
```

**Comment by mkaatman** [comment]
Greg confirmed on SO. http://stackoverflow.com/questions/36625340/phpcbf-complains-about-missing-codesniffer

**Comment by gsherwood** [comment]
> This was probably introduced by #847.

100% correct. Fixed now. Thanks for the reports.

**Comment by NicoCa** [comment]
Is this fix available in the current release?

**Comment by mkasberg** [comment]
The problem is with 2.6.0. I believe the next release, 2.6.1, should include the fix when it becomes available.

**Comment by gsherwood** [comment]
> The problem is with 2.6.0. I believe the next release, 2.6.1, should include the fix when it becomes available.

Correct. I'm trying to fix and review a few more issues before releasing 2.6.1, but I've been sick for the last 2 weeks, so it's going slowly. You can download the 2.5.1 phar is you need a working version more quickly (bottom of this page: https://github.com/squizlabs/PHP_CodeSniffer/releases/tag/2.5.1)

**Comment by Webysther** [comment]
Tested, fixed on 2.6.1. :+1:
